include/h2o/httpclient.h

```diff
@@ -419,6 +419,8 @@ inline int h2o_httpclient__tunnel_is_ready(h2o_httpclient_t *client, int status)
     if (client->upgrade_to != NULL) {
         if (client->upgrade_to == h2o_httpclient_upgrade_to_connect && 200 <= status && status <= 299)
             return 1;
+        if (client->upgrade_to != NULL && 200 <= status && status <= 299 && strcmp(client->upgrade_to, "connect-udp") == 0)
+            return 1;
```

Comment:
We have supported CONNECT-UDP over HTTP/2.

`h2o_httpclient__tunnel_is_ready` is not called simply because there is no need to. In the case of H1, the function has to be called because the framing becomes different than ordinary H1. In the case of H3, it is called because we want to support H3 datagrams. Neither applies to H2.